proposals/0173-Read-Generic-Network-Signal-data.md

```diff
@@ -16,277 +16,306 @@ To enable OEM proprietary mobile apps to read raw CAN/Network data by specifying
+
+* _engineState_ would need to be added to both SDL core and proxy schema. Once new schema is downloaded by both core and proxy, an app may request _engineState_ vehicle data item.
+* It just shows the capability to add a new vehicle data item which OEM can utilize to update the file at cloud. SDL can get updates on this file along with PT update and thus would be able to process new vehicle data
+* Again, each sub items have individual data type and references. So as long as OEM updates the reference file, Mobile apps can access the new data.
+
+### Also, this approach can also be used to read entire 8 bytes CAN message itself
+
+Sample value for map:
+```json
+{
+	"id":"messageName",
+	"dataType":"String",
+	"reference":"OEM_REF_MSG",
+	"array":false,
+	"mandatory":true
+}
+```
+Sample response from module:
+```json
+{
+	"messageName": "AB 04 D1 9E 84 5C B8 22"
+}
+```
+* Data type is String so that it can accommodate a byte string for CAN message. OEM_REF_MSG points to a message definition in OEM mapping table.
+### Integer, Float, String can use the common range values defined in HMI/Mobile API
+### Access to data items would still be controlled using Policies the same way. OEM would need to add new vehicle data items to policies.
 
-    <param name="info" type="String" maxlength="1000" mandatory="false" platform="documentation">
-      <description>Provides additional human readable info regarding the result.</description>
-    </param>
+## Impact on existing code
+* SDL core would need to add support to download and parse the new JSON schema for vehicle data. We can either keep the existing vehicle data items in HMI/Mobile API and only use this approach for new vehicle data items. Or totally replace the way vehicle data items are compiled and validated by SDL. Which is Instead of using the hard coded _xml_ file, valid vehicle data type would be fetched from mapping file over the cloud. (there would be a local copy for intial build). Interface between SDL and HMI need to be updated while passing _GetVehicleData/SubscribeVehicleData/UnsubscribeVehicleData/OnVehicleData_ requests to include return data type and reference key.
 
-    <param name="gps" type="GPSData" mandatory="false">
-      <description>See GPSData</description>
-    </param>
-        
-	<!-- Not showing other params -->
-	
-	<param name="networkData" type="NetworkData" mandatory="false" minsize="1" maxsize="500" maxLength="200" array="true">
-      <description>
-      	Returned Network Data value for a paramId
-      </description>
-    </param> 
+* SDL Proxy needs to be able to download the new JSON schema for vehicle data and add a manager to allow requests for new vehicle data items in a systematic way.
 
-</function>
+* APIs would need a new Request/Response API to act as Generic Request Response for _GetVehicleData/SubscribeVehicleData/UnsubscribeVehicleData_
```

Comment:
Change to: 
APIs would need a new Request/Response API to act as **the** Generic Request Response for _GetVehicleData/SubscribeVehicleData/UnsubscribeVehicleData_

Reply:
Update this please